am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/RevokeAccessIntegrationTest.java

```diff
@@ -27,18 +39,109 @@ void setupTest() {
+    void whenRevokingResourceAccessShouldRemoveFromDatabase() {
         ams.grantExplicitResourceAccess(createGrantForWholeDocument(resourceId, READ_PERMISSION));
         ams.revokeResourceAccess(createMetadata(resourceId));
 
         assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(0);
     }
 
+    @Test
+    void whenRevokingResourceAccessOnRootShouldRemoveFromDatabase() {
+        ams.grantExplicitResourceAccess(createGrantForWholeDocument(resourceId, READ_PERMISSION));
+        revokeResourceAccess("");
+
+        assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(0);
+    }
+
+    @Test
+    void whenRevokingResourceAccessOnSingleNestedAttributeShouldRemoveFromDatabase() {
+        grantExplicitResourceAccess(resourceId, "/test");
+        grantExplicitResourceAccess(resourceId, "/test/childTest");
+        revokeResourceAccess("/test");
+
+        assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(0);
+    }
+
+    @Test
+    void whenRevokingResourceAccessOnMultipleNestedAttributesShouldRemoveFromDatabase() {
+        grantExplicitResourceAccess(resourceId, "/test/childTest/secondChild/thirdChild");
+        revokeResourceAccess("/test/childTest");
+
+        assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(0);
+    }
+
+    @Test
+    void whenPermissionsOnlyOnChildAttributeRevokingPermissionsOnParentShouldCascade() {
+        grantExplicitResourceAccess(resourceId, "/test/childTest");
+        revokeResourceAccess("/test");
+
+        assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(0);
+    }
+
+    @Test
+    void whenPermissionRevokedFromRootShouldDeleteAllChildAttributes() {
+        grantExplicitResourceAccess(resourceId, "/child");
+        grantExplicitResourceAccess(resourceId, "/childTest");
+        grantExplicitResourceAccess(resourceId, "/test/childTest");
+        revokeResourceAccess("");
+
+        assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(0);
+    }
+
+    @Test
+    void whenRevokingSpecificEntryShouldRemoveCorrectEntry() {
+        grantExplicitResourceAccess(resourceId, "/test/childTest");
+        grantExplicitResourceAccess("resource2", "/test/childTest");
+        revokeResourceAccess("/test/childTest");
+
+        assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(0);
+        assertThat(databaseHelper.countExplicitPermissions("resource2")).isEqualTo(1);
+    }
+
+    @Test
+    void whenRevokingAccessOnAttributeShouldRemoveOnlySpecifiedAttributeAndChildren() {
+        grantExplicitResourceAccess(resourceId, "/amount");
+        grantExplicitResourceAccess(resourceId, "/amount/lastUpdated");
+        grantExplicitResourceAccess(resourceId, "/amountInPounds");
+
+        revokeResourceAccess("/amount");
+
+        assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(1);
```

Comment:
looking through this again, I would like an assertion that the attribute stored has the value /amountInPounds rather than anything else. Asserting 1 entry left probably isn't enough. I don't know 100% that this is working as intended.